velox/functions/prestosql/tests/DateTimeFunctionsTest.cpp

```diff
@@ -210,31 +211,34 @@ class DateTimeFunctionsTest : public functions::test::FunctionBaseTest {
     return DATE()->toDays(dateStr);
   }
 
-  RowVectorPtr makeTimestampWithTimeZoneVector(
-      int64_t timestamp,
-      const char* tz) {
+  VectorPtr makeTimestampWithTimeZoneVector(int64_t timestamp, const char* tz) {
     const int64_t tzid = util::getTimeZoneID(tz);
 
-    return std::make_shared<RowVector>(
-        pool(),
-        TIMESTAMP_WITH_TIME_ZONE(),
-        nullptr,
-        1,
-        std::vector<VectorPtr>(
-            {makeNullableFlatVector<int64_t>({timestamp}),
-             makeNullableFlatVector<int16_t>({tzid})}));
+    return makeNullableFlatVector<int64_t>(
+        {pack(timestamp, tzid)}, TIMESTAMP_WITH_TIME_ZONE());
   }
 
-  RowVectorPtr makeTimestampWithTimeZoneVector(
+  VectorPtr makeTimestampWithTimeZoneVector(
       const VectorPtr& timestamps,
       const VectorPtr& timezones) {
```

Comment:
I wonder how many callers are there of this utility function. Could we let this function receives two functors like the following instead?
```
VectorPtr makeTimestampWithTimeZoneVector(
std::function<int64_t(vector_size_t row)> timestampAt,
std::function<int16_t(vector_size_t row)> timezoneAt);
```

Reply:
Refactored.